module/documents/actor/actor.mjs

```diff
@@ -3039,8 +3041,12 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
       tokenData.elevation = this.token.elevation;
       tokenData.hidden = this.token.hidden;
       tokenData.rotation = this.token.rotation;
-      const previousActorData = this.token.delta.toObject();
-      foundry.utils.setProperty(tokenData, "flags.dnd5e.previousActorData", previousActorData);
+      if ( !this.token.flags.dnd5e?.previousActorData ) {
+        const previousActorData = this.token.delta.toObject();
+        foundry.utils.setProperty(tokenData, "flags.dnd5e.previousActorData", previousActorData);
+        foundry.utils.setProperty(tokenData, "flags.dnd5e.previousTokenData.texture.src", this.token.texture.src);
+        foundry.utils.setProperty(tokenData, "flags.dnd5e.previousTokenData.name", this.token.name);
```

Comment:
Looks like there are a few more things beyond just name and texture that get changed during transform: `width`, `height`, `texture.tint`, etc. Presumably we would want to restore those as well.